app/Http/Middleware/AdminMiddleware.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use App\User;
+use Closure;
+
+class AdminMiddleware
+{
+    /**
+     * declaring constant signifying to compare to the role_id
+     *
+    */
+    const REGULAR_USER       = 1;
+    const ADMIN_USER         = 2;
+    const SUPER_ADMIN_USER   = 3;
+
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        $token = $request->query('token');
+
+        if (!empty($token)) {
+            $appToken = User::where('api_token', '=', $token)
+                ->first();
+
+            if (is_null($appToken)) {
+                return response()->json(['message' => 'User unauthorized due to invalid token'], 401);
```

Comment:
Status codes and error messages could be moved in another file, say ...

```
// StatusCode.php

const UNAUTHORISED = 401;
const CREATED = 201;
const OK = 200;
```

```
ErrorMessages/Users.php or ErrorMessages/Authentication.php

const UNAUTHORISED = User unauthorized due to invalid token.
```

You should end up with something like .....

`return response()->json(['message' => errorMessage.UNAUTHORISED], statusCode.UNAUTHORISED);`